File: samples/java17-sample-contract/src/main/java/my/sample/java17/contract/rockps/MainDispatch.java
package my.sample.java17.contract.rockps;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.json.JsonMapper;
import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
import com.wavesenterprise.sdk.contract.core.dispatch.ContractDispatcher;
import com.wavesenterprise.sdk.contract.grpc.GrpcJacksonContractDispatcherBuilder;

public class MainDispatch {
    public static void main(String[] args) {
        ContractDispatcher contractDispatcher = GrpcJacksonContractDispatcherBuilder.builder()
                .contractHandlerType(RockPaperScissorsContractImpl.class)
                .objectMapper(getObjectMapper())
                .build();

        contractDispatcher.dispatch();
    }

    private static ObjectMapper getObjectMapper() {
        return JsonMapper.builder().addModule(new JavaTimeModule()).build();
    }
}

File: samples/kotlin-sample-contract/src/main/kotlin/my/sample/kotlin/contract/rockps/MainDispatch.kt
package my.sample.kotlin.contract.rockps

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.databind.json.JsonMapper
import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
import com.wavesenterprise.sdk.contract.grpc.GrpcJacksonContractDispatcherBuilder

class MainDispatch {
    fun main(args: Array<String>) {
        val contractDispatcher = GrpcJacksonContractDispatcherBuilder.builder()
            .contractHandlerType(RockPaperScissorsContractImpl::class.java)
            .objectMapper(objectMapper)
            .build()
        contractDispatcher.dispatch()
    }

    private val objectMapper: ObjectMapper
        get() = JsonMapper.builder().addModule(JavaTimeModule()).build()
}